src/rotator_library/distributed_gate.py

```
@@ -0,0 +1,382 @@
+
+    # --- ConcurrencyGate (config-less => defaults: max=2, interval=0) ---
+    gate = ConcurrencyGate(config_path="/nonexistent/policy.yaml")
+    assert gate.try_acquire("p", "m", now=now) is True   # 1/2
+    assert gate.try_acquire("p", "m", now=now) is True   # 2/2
+    assert gate.try_acquire("p", "m", now=now) is False  # full -> fall back
+    # a different model on the same provider has its own slots
+    assert gate.try_acquire("p", "other", now=now) is True
+    gate.release("p", "m")
+    assert gate.try_acquire("p", "m", now=now) is True   # freed -> 2/2 again
+    # release is idempotent-safe (never goes negative)
+    gate.release("p", "m"); gate.release("p", "m"); gate.release("p", "m")
```

Ruff (0.15.18)

[error] Multiple statements on one line (semicolon)

(E702)